### PR TITLE
Fix duplicated image markup fields when registration fails

### LINKED ISSUE #83: Duplicate image markup if registration form validation fails
image

> When im registering a new user, and i have uploaded the profile picture but the password is not strong enough when the page refresh, it duplicates the image preview markup, but broken

### Steps to recreate

- Install the plugin
- Add the avatar image field to the field of the default registration form
- Add the password field field to the field of the default registration form
- Visit the registration formed when logged out (or incognito)
- Fill out the form, uploading an image, using a weak password
- Get form validation error about password and see duplicate avatar image markup

### PR DESCRIPTION
**Pull request by polevaultweb**
Resolves #83

### CODE REVIEW

**Comment by polevaultweb** [comment]
@singhrvijay23 this doesn't actually fix it for me.

The line here which checks if the `$data->value` is an array is actually for if the file field has multiple files uploaded (which isn't supported as yet) https://github.com/WPUserManager/wp-user-manager/blob/233753b30bef26dc7c7ea24a2e31ba2ede9aebaf/templates/form-fields/file-field.php#L35

Therefore we need to check if the `$data->value` is an associative array (ie. with the key 'url') and if so don't loop over the items, instead just pass the `$data->value['url']` as the value to the template (like is done in the elseif part).

Hope that makes sense, let me know if you need anything else.

File: templates/form-fields/file-field.php [hunk 1]
```diff
@@ -32,19 +32,20 @@
 <div class="wpum-uploaded-files">
 	<?php if ( ! empty( $data->value ) ) : ?>
 		<?php if ( is_array( $data->value ) ) : ?>
-			<?php foreach ( $data->value as $value ) : ?>
+			<?php 	if (isset($data->value['url']) || array_key_exists('url', $data->value)) {
+ 				?>
```
Comment (polevaultweb) [inline templates/form-fields/file-field.php:36]: Might as well remove this line, and the <?php in the line below

File: templates/form-fields/file-field.php [hunk 2]
```diff
@@ -32,19 +32,20 @@
...
 						->set_template_data( [
 							'key' => $data->key,
 							'name' => 'current_' . $field_name,
-							'value' => $value,
+							'value' => $data->value['url'],
 							'type'  => $data->type,
 							'field' => []
 						] )
 						->get_template_part( 'form-fields/file', 'uploaded' );
 				?>
```
Comment (polevaultweb) [inline templates/form-fields/file-field.php:47]: Same here, clean up closing and opening PHP tags

File: templates/form-fields/file-field.php [hunk 3]
```diff
@@ -32,19 +32,20 @@
 <div class="wpum-uploaded-files">
 	<?php if ( ! empty( $data->value ) ) : ?>
 		<?php if ( is_array( $data->value ) ) : ?>
-			<?php foreach ( $data->value as $value ) : ?>
+			<?php 	if (isset($data->value['url']) || array_key_exists('url', $data->value)) {
```
Comment (polevaultweb) [inline templates/form-fields/file-field.php:35]: Please keep the formating to the existing style (based on https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/) - maybe your code editor / IDE can autoformat for you

Comment (polevaultweb) [inline templates/form-fields/file-field.php:35]: Why do an isset or array_key_exists?

Comment (polevaultweb) [reply templates/form-fields/file-field.php]: The code formatting is more than just line alignment, its spaces between functions and parenthesis etc.

Comment (polevaultweb) [reply templates/form-fields/file-field.php]: @singhrvijay23 the isset is enough. The call to array_key_exists will only be reached if the url key is present but has a null value, in which case there is no need to check again if the key exists

**Comment by singhrvijay23** [comment]
Will take care of this in future tasks.


Regards,
Ran Vijay Singh
7007412242
-----------------------------------------------------------------------------------------------------------------

**Pull request merged**